test/test_linalg.py

```diff
@@ -7514,6 +7514,31 @@ def test():
         a = torch.ones(300)
         b = torch.ones(300)
         check_correctness(torch.dot, a, b)
+    
+    @dtypes(*itertools.product(floating_and_complex_types(), floating_and_complex_types()))
+    def test_baddbmm_mixed_dtypes_backward(self, device, dtypes):
+        # test that baddbmm can backward successfully when input tensors have different dtypes
```

Comment:
Nice testing! 
I'm curious if baddmm errors when a and b have different dtypes. (If so, we could make dtypes contain 3 dtypes, and assert that those cases error? if not, we'd want to test those cases?)

Reply:
baddbmm will actually fail if a and b have different dtypes.
This is as baddbmm performs a batch matrix-matrix product of a and b. m is added to the final result.
Torch has dtype promotion for matrix addition but not matrix product for some reason.

Reply:
Oh ok, thanks - a comment about that in the test would also sound good